### ISSUE TITLE
group topic defacs changes are not propagated to group members

### ISSUE THREAD

**Issue by gabriel-vasile** [tinode/chat#720]
I'm not sure if this is a missing feature or this is the intended behaviour.

The situation:
- a group topic with 2 members (owner and member)
- owner changes defacs for the group

What I expected: the change to be sent to group member.
What happens: group member receives the new defacs only at the next {sub} to the group topic.

Tinode version: master

**Comment by or-else** [comment]
I would say it's probably a bug but a very minor one. 
Can you describe a use case when such update is important?

**Comment by gabriel-vasile** [comment]
The other changes made by the owner to the group (changes in public field, changes to the group members list) are propagated and peers views are updated in real-time. 
Not the same for defacs...

One can argue defacs don't change so often and they don't deserve to be updated in real-time, idk

**Comment by or-else** [comment]
I agree that it's inconsistent and that's why I marked it as a bug. 
The only case that I can think of when this bug may case minor trouble is if the owner has more than one device and changes the defacs from one of them then tries to do it again from another.
Do you see any other cases?

**Comment by gabriel-vasile** [comment]
The case we have is the owner changes the defacs and other group members which are online and subscribed to the group are not notified of the change. They still see the old permissions in the Info section, their UI is still working based on the old defacs.

**Comment by or-else** [comment]
I understand that they do not get updated. That's why this issue is marked as a bug.
Why is it a problem that they are not updated (other than being inconsistent)? 
"They want to do X but do Y because of the bug".

**Comment by gabriel-vasile** [comment]
Owner removes J from group permission.
User1 invites user2 to group, knowing the group has J.
User2 cannot join.

This is not a situation in webapp, or the other frontends but this is what I'm facing.
Having the clients receive a notification that permissions have changed will let them handle situation.

**Comment by or-else** [comment]
User2 could not join because the owner removed the J permission, not because User1 had incorrect information.
Anyway, it does indeed look like a minor bug. It will be fixed eventually.
